lean4-infoview/src/infoview/tooltips.tsx

```diff
@@ -83,18 +86,19 @@ export const HighlightOnHoverSpan = forwardAndUseRef<HTMLSpanElement, React.HTML
     // tree and not just a logical React child (see useLogicalDom and
     // https://reactjs.org/docs/portals.html#event-bubbling-through-portals).
     if (ref.current && e.target instanceof Node && ref.current.contains(e.target)) {
-      if ('_HighlightOnHoverSpanSeen' in e) return
-      (e as any)._HighlightOnHoverSpanSeen = {}
-      setIsPointerOver(b)
+      if ('_DetectHoverSpanSeen' in e) return
+      (e as any)._DetectHoverSpanSeen = {}
+      if (!b) setHoverState('off')
```

Comment:
Oh, never mind, it's because it's over here.  Hmm.  Its a bit messy that DetectHoverSpan is involved in the setHoverState book keeping along with InteractiveCode.tsx.  Why wouldn't WithTooltipOnHover do all the other setHoverState bookkeeping instead of doing it in InteractiveCode.tsx that way localizing it all to ToolTip.tsd ?  The only thing InteractiveCode really needs is the onClick to call ec.revealPosition, no?

Reply:
So some level of interaction between `InteractiveCodeTag` and the tooltips.tsx components will always be required because the former holds the go-to implementation, and importantly we need to trigger a state update in the `InteractiveCodeTag` in order to re-render it. But you're right that the event handlers didn't have to be there, so I moved them to `DetectHoverSpan`.

Also, we could in principle do everything in `WithTooltipOnHover` and delete `DetectHoverSpan`, but it would become a monster component so I rather keep them separate but smaller.

Reply:
thanks, this is better, but I think you broke ctrl key in infoview in the process, because it when Tooltip calls `setHoverState(st => st === 'over' ? 'ctrlOver' : st)` it's not calling the real setHoverState it's calling this:
```
        setHoverState={st => {
          // On ctrl-hover, fetch the go-to location
          if (st === 'ctrlOver') void fetchGoToLoc()
          setHoverState(st)
        }}
```
and so the check ` if (st === 'ctrlOver')` desn't work when st is a lambda (st => st === 'over' ? 'ctrlOver' : st) and you can't use `hoverState` here, so perhaps you have to do this:

```
      // On ctrl-hover, fetch the go-to location
      // process the given st which could be a lambda
      setHoverState(st);
      // now use setHoverState again to query the resulting new state.
      setHoverState(st => {
          if (st === 'ctrlOver')
              void fetchGoToLoc();
          return st;
      });
```
but this doesn't fully fix it either...

Reply:
I did indeed, thanks!

Reply:
still having trouble, I've had your lean4 build installed and everything working before, I just updated your lean4 bits, and ctrl+click works fine in text editor but in the info view I get an exception saying `Lean.Widget.getGoToLocation` doesn't exist... which is odd because I do see it in your repo in `C:\msys64\home\clovett\git\vtec234\lean4\src\Lean\Server\FileWorker\WidgetRequests.lean`
perhaps I have a setup error... I'm using `elan toolchain link master C:\msys64\home\clovett\git\vtec234\lean4/build/release/stage1` which is what I did before when everything was working.... weird.

Reply:
Never mind, I was in a multi-workspace and forgot to change one of my lean-toolchains to point to the right version.  It works now.  But I did get one print out early on while lean was still loading I guess:
```
Error in go-to-definition:  {"0":"C","1":"l","2":"i","3":"e","4":"n","5":"t","6":" ","7":"i","8":"s","9":" ","10":"n","11":"o","12":"t","13":" ","14":"r","15":"u","16":"n","17":"n","18":"i","19":"n","20":"g","length":21}
```
but it started working after this happened.